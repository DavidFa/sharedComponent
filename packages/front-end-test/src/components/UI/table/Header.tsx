import React from 'react'
import TableCell from './TableCell';
import Row from './Row';
import styled from 'styled-components'

const THead = styled.thead`
display: table-header-group;
vertical-align: middle;
border-color: inherit;
`;

type HeaderProps = {
    headers: string[];
}

const Header: React.FC<HeaderProps> = ({ headers }) => {

    const fields = !!headers.length && <Row>{headers.map((item, index) => {
        return <TableCell key={`${item}_${index}`}>{item}</TableCell>
    })}</Row>;

    return (
        <THead>
            {fields}
        </THead>
    )
}

export default Header;
