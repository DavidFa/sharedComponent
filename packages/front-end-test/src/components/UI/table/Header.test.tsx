import Header from "./Header";
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';

describe("Header", () => {
  const headers: string[] = ["id", "title", "action"];

  describe("Snapshot Test", () => {
    it("should render with 3 td", () => {
      const component = renderer.create(<Header headers={headers} />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should render without  td", () => {
      const component = renderer.create(<Header headers={[]} />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  describe("Dom Test", () => {
    const table = document.createElement("table");
    const tableContainer = document.body.appendChild(table);

    it("should contains a header with tr, and 3 td", () => {
      const { container } = render(<Header headers={headers} />, {
        container: tableContainer
      });

      expect(container.firstChild).toMatchInlineSnapshot(`
        <thead
          class="sc-hKgJUU BiQOs"
        >
          <tr
            class="sc-dlfnuX bsBZvW"
          >
            <td
              class="sc-bdfBQB bjKVnb"
              role="cell"
            >
              id
            </td>
            <td
              class="sc-bdfBQB bjKVnb"
              role="cell"
            >
              title
            </td>
            <td
              class="sc-bdfBQB bjKVnb"
              role="cell"
            >
              action
            </td>
          </tr>
        </thead>
      `);

      const rowgroup = screen.queryByRole("rowgroup");
      expect(rowgroup).toBeInTheDocument();

      const row = screen.queryByRole("row");
      expect(row).toBeInTheDocument();

      const cells = screen.queryAllByRole("cell");
      expect(cells).toHaveLength(3);

      const id = screen.queryByText("id");
      expect(id).toBeInTheDocument();

      const title = screen.queryByText("title");
      expect(title).toBeInTheDocument();

      const action = screen.queryByText("action");
      expect(action).toBeInTheDocument();
    });

    it("should contains a header without tr and td", () => {
      const { container } = render(<Header headers={[]} />, {
        container: tableContainer
      });

      expect(container.firstChild).toMatchInlineSnapshot(`
        <thead
          class="sc-hKgJUU BiQOs"
        />
      `);

      const row = screen.queryByRole("row");
      expect(row).not.toBeInTheDocument();

      const cells = screen.queryAllByRole("cell");
      expect(cells).toHaveLength(0);

      const id = screen.queryByText("id");
      expect(id).not.toBeInTheDocument();

      const title = screen.queryByText("title");
      expect(title).not.toBeInTheDocument();

      const action = screen.queryByText("action");
      expect(action).not.toBeInTheDocument();
    });
  });
});
