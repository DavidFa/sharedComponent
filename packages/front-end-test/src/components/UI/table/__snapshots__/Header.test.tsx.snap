// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Test should render with 3 td 1`] = `
<thead
  className="sc-hKgJUU BiQOs"
>
  <tr
    className="sc-dlfnuX bsBZvW"
  >
    <td
      className="sc-bdfBQB bjKVnb"
      role="cell"
    >
      id
    </td>
    <td
      className="sc-bdfBQB bjKVnb"
      role="cell"
    >
      title
    </td>
    <td
      className="sc-bdfBQB bjKVnb"
      role="cell"
    >
      action
    </td>
  </tr>
</thead>
`;

exports[`Header Snapshot Test should render without  td 1`] = `
<thead
  className="sc-hKgJUU BiQOs"
/>
`;
